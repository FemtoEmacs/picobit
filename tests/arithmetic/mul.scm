(define-macro (test-cond condition name) ;; TODO watch out for name capture
  `(let ((DUMMY-x ,condition))
     (display ,name)
     (if DUMMY-x
       (display " : PASSED\n")
       (begin (display " : FAILED, got : ")
	      (display DUMMY-x)
	      (display "\n")))))

(test-cond (= (* 2 2) 4) "2 * 2 = 4")
(test-cond (= (* 0 5) 0) "0 * 5 = 0")
(test-cond (= (* 2 7) 14) "2 * 7 = 14")
(test-cond (= (* 1 -5) -5) "1 * -5 = -5")
(test-cond (= (* 2 -5) -10) "2 * -5 = -10")
(test-cond (= (* -7 5) -35) "-7 * 5 = -35")
(test-cond (= (* -3 -7) 21) "-3 * -7 = 21")

(test-cond (= (* 10000 10000) 100000000) "1e4 * 1e4 = 1e8")
(test-cond (= (* -10000 10000) -100000000) "-1e4 * 1e4 = -1e8")
(test-cond (= (* 10000 -10000) -100000000) "1e4 * -1e4 = -1e8")
(test-cond (= (* -10000 -10000) 100000000) "-1e4 * -1e4 = 1e8")

(test-cond (= (* 100000 10000) 1000000000) "1e5 * 1e4 = 1e9")
(test-cond (= (* 10000 100000) 1000000000) "1e4 * 1e5 = 1e9")
(test-cond (= (* 100000 100000) 10000000000) "1e5 * 1e5 = 1e10")
;; from that we can infer that the neg function (that negates, obviously) works
;; since it is used by multiplication when operands are negative
