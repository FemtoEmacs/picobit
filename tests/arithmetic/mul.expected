:0080000080
:10800000FBD7F20381A44000813A4000812F400059
:10801000000427100003D8F0000586A00003FFFB32
:108020000003FFF901AEE10001D4CA0001381F00CE
:10803000818F4000819C400081A3400081AB4000C3
:1080400081B8400001BBE4000003FFFD81D840007F
:1080500081E9400081ED400081EE400081F2400066
:108060000003FFDD816C400081724000817B400095
:1080700081834000813F40008144400081494000AD
:10808000814D4000813E40000003FFF68035018DA8
:108090008040018E80690190802401918048019286
:1080A000804101938041019480400195802E01968A
:1080B0008024019780240198803501998024019AB9
:1080C0008024019B8041019D8024019E8038019F76
:1080D000803B01A00003FA0A803801A1802401A29C
:1080E000804801A4802401A5803D0002803601A6BD
:1080F000803101A7806601A8802E01A9802401AAF1
:10810000807301AC803601AD803C0002806901AF14
:10811000802401B0804501B1803101B2804101B3BA
:10812000802401B4803501B5802701B68039000272
:10813000806901B7803801B9803501BA803101BC4E
:10814000804501BD802401BE803201BF804101C055
:10815000803901C1803601C2803801CC807D01C3E5
:10816000803B01C4802401C5802401C6802401C74E
:108170008031014E803501C8806601C9803101CA55
:10818000802401CB8024014C803901CC802401CD96
:10819000802401CE80780002803101CF802401D07C
:1081A000802E01D1803501D2804101D3802E01D5AE
:1081B000802401D6802401D7803B01D9803501DAA3
:1081C000802E01DB806901DC807001DD803901DEF9
:1081D000806901DF803501E0802401E1802E01E229
:1081E000803501E3807801DD802401E4805701E5DA
:1081F000802401E6803501E780240184803801E88D
:10820000803601EA80340002806901EB802E01ECA7
:10821000806901EF805001F0803901F1803601F36F
:10822000802E01F4804101F5806901598077015A5F
:108230008031015B8039015C8041015D8030015EED
:108240008024015F802401608073016180240162C9
:10825000803E0163802E016480690165802E016685
:10826000802E016780270168802401698041016AAE
:108270008024016B8024016D8024016E803E016F9B
:1082800080350170800E00028031017180240173FD
:108290008035017480730175802401768035017703
:1082A00080310178807801798024017A000405F515
:1082B0008067017C8041017D8057017E8037017F8E
:1082C000802401808041018180690182806A0002EE
:1082D0008035018480350185802401868038000244
:1082E0000006540B80350187804D0188803101895B
:1082F000802400028024018A8024018B8024018C48
:108300008071014080240141802E01428031016052
:10831000806B014380690145806E014680390147C9
:10832000805401488024014A8039014B8031014C3E
:108330008037014E8035014F802401BC8069015097
:108340008024015100043B9A80240152804A015349
:10835000802E0154802401558024015680690157E4
:108360008024015880380133804200028024013487
:108370008039013580690136803E01BF8024013794
:1083800080690139803101B78049013B802E013C71
:108390008069013D8024012C8027012D8024012E3D
:1083A0008038013080240131803401328035012849
:1083B000803801298049012A8024012B8027012748
:1083C0008035000280240126802401BA02B407947B
:1083D0005002B407155102B4070F52060640620856
:1083E000C702B406F161EE0409406204C702B40694
:1083F000D361EE060B406212C702B406B561EE050A
:10840000A10A4062A10AC702B4069761EE06A10A5A
:108410004062A125C702B4067961EEA10B09406252
:10842000A11BC702B4065B61EEA115A10B40621946
:10843000C702B4063D61EEA107A1074062A10CC7C7
:1084400002B4061F61EEA108A1074062A10EC70297
:10845000B4060161EEA107A1084062A10EC702B4F3
:1084600005E361EEA108A1084062A10CC702B405B2
:10847000C561EEA109A1074062A10DC702B405A71D
:1084800061EEA107A1094062A10DC702B40589618F
:10849000EEA109A1094062A114C702B404A161EED2
:1084A000D801A119B589EE20B30579A106B68001DE
:1084B00020DAB8954221DBB6800221CC9A21CE2168
:1084C00061EE2022CFB67200EF20B6800120DA9E46
:1084D000A02607E620B558EEA02607E600EF20C04C
:1084E000B8BF20B582B648012004C8B8AEA031217B
:1084F000B5A402B583CDDCEF02A034210EC5C122A4
:10850000CD2102CDB405087101210EC89220EF21C2
:108510000E416221B662010421C2EF2002B559B7B3
:108520005520CCB8B8A02C07E620CEB51FEE20CF42
:10853000B6800120D296A02D07E600EF20CC9DA0AA
:108540002407E620CEB505EE20CFB666A116B0040E
:10855000AFEE20B004CCEEA02D07E600EF20D995B9
:10856000A11AB104AF20F09C2095A110B104AFA1D5
:1085700023B104AFA117B104AFA105B004AFEE2041
:10858000B004AFEEA104B104AF01A11CB004AFEE82
:108590002095A106B104AFA105B004AFEE20B00450
:1085A000AFEEA104B104AF01A11DB004AFEE209560
:1085B000A106B104AFA105B004AFEE20B004AFEE48
:1085C000A104B104AF01A120B004AFEE2095A10633
:1085D000B104AFA105B004AFEE20B004AFEEA1042A
:1085E000B104AF01A11EB004AFEE2095A106B10405
:1085F000AFA105B004AFEE20B004AFEEA104B1040A
:10860000AF01A112B004AFEE2095A106B104AFA155
:1086100005B004AFEE20B004AFEEA104B104AF0189
:10862000A113B004AFEE2095A106B104AFA105B02F
:1086300004AFEE20B004AFEEA104B104AF01A1225B
:10864000B004AFEE2095A106B104AFA105B004AF10
:10865000EE20B004AFEEA104B104AF01A10FB0044D
:10866000AFEE2095A106B104AFA105B004AFEE2096
:10867000B004AFEEA104B104AF01A11FB004AFEE8E
:108680002095A106B104AFA105B004AFEE20B0045F
:10869000AFEEA104B104AF01A111B004AFEE20957B
:1086A000A106B104AFA105B004AFEE20B004AFEE57
:1086B000A104B104AF01A121B004AFEE2095A10641
:1086C000B104AFA105B004AFEE20B004AFEEA10439
:1086D000B104AF01A118B004AFEE2095A106B1041A
:1086E000AFA105B004AFEE20B004AFEEA104B10419
:1086F000AF01A124B004AFEE2095A106B104AFA153
:1087000005B004AFEE20B004AFEEA104B104AF0198
:108710002007E600EF022004C8212302CDCDB407D4
:108720002271012204C822242302CDCDCDB4073109
:10873000710123B8DC21B0051623222602CDCDCD50
:10874000B40744710122B8C621B0051621242602BF
:10875000CDCDCDB4075771012022242602CDCDCD39
:10876000CDB4076570002121C4232502CDCDB40707
:1087700072710122B893219220EF2298019420B1C6
:10878000051620EF21B87BB77500B86EB76A21B720
:108790003B21B725FE21CCB3082E2022B680022132
:1087A000CCB3082C2022CEB58422CFB6710220048F
:1087B000C8212302CDCDB407BA71012204C82224F6
:1087C0002302CDCDCDB407C9710123B8DC21B0059A
:1087D0001623222602CDCDCDB407DC710122B8C606
:1087E00021B0051621242602CDCDCDB407EF7101AD
:1087F0002022242602CDCDCDCDB407FD700021214D
:10880000C3232502CDCDB4080A710122B893219269
:1088100020EF2298019420B1051620EF21B87BB7F4
:108820007500B86EB76A21B73B21B72520EF20EF5E
:00000001FF
